Neos.Neos/Classes/Domain/Service/WorkspaceService.php

```diff
@@ -122,44 +132,64 @@ public function createRootWorkspace(ContentRepositoryId $contentRepositoryId, Wo
                 ContentStreamId::create()
             )
         );
-        $this->metadataAndRoleRepository->addWorkspaceMetadata($contentRepositoryId, $workspaceName, $title, $description, WorkspaceClassification::ROOT, null);
-    }
-
-    /**
-     * Create the "live" root workspace with the default role assignment (users with the role "Neos.Neos:LivePublisher" are collaborators)
-     */
-    public function createLiveWorkspaceIfMissing(ContentRepositoryId $contentRepositoryId): void
-    {
-        $contentRepository = $this->contentRepositoryRegistry->get($contentRepositoryId);
-        $workspaceName = WorkspaceName::forLive();
-        $liveWorkspace = $contentRepository->findWorkspaceByName($workspaceName);
-        if ($liveWorkspace !== null) {
-            // live workspace already exists
-            return;
+        $this->metadataAndRoleRepository->addWorkspaceMetadata($contentRepositoryId, $workspaceName, $title, $description, WorkspaceClassification::ROOT, ownerUserId: null);
+        foreach ($assignments as $assignment) {
+            $this->metadataAndRoleRepository->assignWorkspaceRole($contentRepositoryId, $workspaceName, $assignment);
         }
-        $this->createRootWorkspace($contentRepositoryId, $workspaceName, WorkspaceTitle::fromString('Public live workspace'), WorkspaceDescription::empty());
-        $this->metadataAndRoleRepository->assignWorkspaceRole($contentRepositoryId, $workspaceName, WorkspaceRoleAssignment::createForGroup('Neos.Neos:LivePublisher', WorkspaceRole::COLLABORATOR));
     }
 
     /**
-     * Create a new, personal, workspace for the specified user
+     * Create a new, personal, workspace for the specified user (fails if the user already owns a workspace)
      */
     public function createPersonalWorkspace(ContentRepositoryId $contentRepositoryId, WorkspaceName $workspaceName, WorkspaceTitle $title, WorkspaceDescription $description, WorkspaceName $baseWorkspaceName, UserId $ownerId): void
     {
-        $this->createWorkspace($contentRepositoryId, $workspaceName, $title, $description, $baseWorkspaceName, $ownerId, WorkspaceClassification::PERSONAL);
+        $existingUserWorkspace = $this->metadataAndRoleRepository->findPrimaryWorkspaceNameForUser($contentRepositoryId, $ownerId);
+        if ($existingUserWorkspace !== null) {
+            throw new \RuntimeException(sprintf('Failed to create personal workspace "%s" for user with id "%s", because the workspace "%s" is already assigned to the user', $workspaceName->value, $ownerId->value, $existingUserWorkspace->value), 1733754904);
```

Comment:
isn't that an `InvalidArgumentException` then?

Reply:
If _I_ like? This is really not about me :)
I just commented on it because there seems to be confusion with exception types.

Personally I used `InvalidArgumentException` when the input did not make sense – on the _type_ level but also _logically_. The latter is not correct, I just learned:
![image](https://github.com/user-attachments/assets/a3f44cb2-e759-4c40-9fa7-9c0edda8f94b)

In the end, it does not really matter that much in this case and `RuntimeException` is totally fine